assignments/closure.js

```diff
@@ -1,11 +1,31 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
 
+function sayHello() {
+  var say = function() {
+    console.log(hello);
+  };
+  // Local variable that ends up within the closure
+  var hello = "Hello, world!";
+  return say;
+}
+var sayHelloClosure = sayHello();
+sayHelloClosure(); // ‘Hello, world!’
 
 // ==== Challenge 2: Create a counter function ====
-const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
-};
+// const counter = () => {
+// Return a function that when invoked increments and returns a counter variable.
+var add = (function() {
+  var counter = 0;
+  return function() {
+    counter += 1;
+    return counter;
+  };
+})();
+
+add();
+add();
+add();
 // Example usage: const newCounter = counter();
```

Comment:
In order to create a closure you need to save the function in another variable. For example:

```
const newCounter = add();
console.log(newCounter()) // 1
console.log(newCounter()) // 2
console.log(newCounter()) // 3
console.log(newCounter()) // 4
console.log(newCounter()) // 5
```